lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -823,3 +823,59 @@ Type circt::firrtl::lowerType(Type type) {
 
   return {};
 }
+
+Type circt::firrtl::lowerType(Type type, hw::HWModuleOp module) {
+  hw::TypeScopeOp typeScope;
+  for (auto scopeOp : module.getOps<hw::TypeScopeOp>()) {
+    typeScope = scopeOp;
+    break;
+  }
```

Comment:
It should cause regression especially when lowering normal modules as there is no type scope.

Reply:
I moved this to the global scope, create a single TypeScope and cache it. That should avoid a performance regression.